### ISSUE TITLE
Problem creating new entries with relationships Jelly-0.9.5.1

### ISSUE THREAD

**Issue by raeldc** [jonathangeiger/kohana-jelly#28]
This is the same as Issue #10.

I think it deserves a new issue.

This is my code:

`
        $user = Model::factory('user');
        $login = Jelly::select('role')->where('name', '=', 'login')->load();
        $admin = Jelly::select('role')->where('name', '=', 'admin')->load();

```
    $user->set(array(
        'username' => 'test',
        'password' => 'testuser',
        'password_confirm' => 'testuser',
        'email'   => 'testuser@test.com',
        'roles' => array($login, $admin),
        'logins' => 0,
    ));

    $user->save();
```

`

This is the error message:

```
Database_Exception [ 1452 ]: Cannot add or update a child row: a foreign key constraint fails (`kohana3`.`roles_users`, CONSTRAINT `roles_users_ibfk_2` FOREIGN KEY (`role_id`) REFERENCES `roles` (`id`) ON DELETE CASCADE) [ INSERT INTO `roles_users` (`role_id`, `user_id`) VALUES (NULL, 1), (NULL, 2) ]
```

**Comment by raeldc** [comment]
While you're at it you'll also notice that there's something wrong with how the query was built.

Look closer:

```
INSERT INTO `roles_users` (`role_id`, `user_id`) VALUES (NULL, 1), (NULL, 2)
```

Look at the ordering of 

```
(`role_id`, `user_id`) 
```

and `VALUES (NULL, 1), (NULL, 2)`. The second column is `user_id` but in the insertion of values, the second column is the role_id in this case, 1 and 2.

**Comment by jonathangeiger** [comment]
I believe this is fixed with 921360d0de1a8805cd35d8077609969433c21cdf. Can you confirm?

**Comment by raeldc** [comment]
It works! Thanks Jon!
